search-app/src/cmr/search/services/association_validation.clj

```diff
@@ -481,7 +481,7 @@
   (let [concept-ids (concat [(:concept-id concept)] (map :concept-id associations))
         concept-types (map #(concepts/concept-id->type %) concept-ids)
         supported-concept-types (set (concat [:collection :variable :service :tool]
-                                             (concepts/get-generic-concept-types-array)))
+                                             concepts/get-generic-non-draft-concept-types-array))
```

Comment:
I do think its worthwhile to add that as a note in that section to avoid a later confusion just something like: `Note: Draft records cannot be associated to any concepts.`